jdbc/src/main/java/org/apache/metamodel/jdbc/dialects/HiveQueryRewriter.java

```diff
@@ -35,6 +35,10 @@ public String rewriteColumnType(ColumnType columnType, Integer columnSize) {
         if (columnType == ColumnType.INTEGER) {
             return "INT";
         }
+        // Hive does not support VARCHAR without a width, nor VARCHAR(MAX).
+        if (columnType == ColumnType.STRING && columnSize == null) {
```

Comment:
Maybe checking with STRING isn't quite enough here. I could imagine a lot of Hive users using the more specific VARCHAR column type.